### ISSUE TITLE
Pods are not receiving a balanced load

### ISSUE THREAD

**Issue by formuzi** [miekg/coredns#6754]
**What happened**:
We are updating coredns from version 1.9.3 to version 1.10.1. Following the upgrade, we saw an imbalance in traffic where two pods received less traffic and one pod received the most query requests.
<img width="1674" alt="Screenshot 2024-06-20 at 2 54 42 PM" src="https://github.com/coredns/coredns/assets/38672652/813f0315-00e2-40c7-a717-ba9cd3c6e0e8">

**What you expected to happen**:
There should be an balanced distribution of requests throughout all pods.

**How to reproduce it (as minimally and precisely as possible)**:
Use v1.10.1 for coredns deployment and check request count for each pod

**Anything else we need to know?**:
I think the issue was introduced by this PR (https://github.com/coredns/coredns/pull/5784) . It might have been caused by removing the query limit per connection.
Here is request count difference after revert above PR
![revert_pr](https://github.com/coredns/coredns/assets/38672652/ebc72f77-2129-42eb-81f0-0ae6c3ed7045)

**Environment**:

- the version of CoreDNS: v1.10.1
- Corefile:
- logs, if applicable:
- OS (e.g: `cat /etc/os-release`):
- Others:
Kubernetes version: v1.27.13

**Comment by YeshuGarg1** [comment]
Did you find the cause of this? I am also experiencing the same

**Comment by selin6** [comment]
@YeshuGarg1 This is caused by removing the query limit per connection. 
The easy solution is to change the dns.Server's MaxTCPQueries setting back to its default value. 
<img width="762" alt="Screenshot 2024-11-17 at 2 30 21 PM" src="https://github.com/user-attachments/assets/75763f47-88be-43ca-b4d4-781213fba70a">

**Comment by xlagg5** [comment]
do you know If I can change the MaxTCPQuerries setting in my coredns configmap for k8s?

**Comment by selin6** [comment]
No, the MaxTCPQuerries, which use a fixed number, cannot be changed according to the [code](https://github.com/coredns/coredns/blob/master/core/dnsserver/server.go#L436)

**Comment by timetofaceit** [comment]
Same here... Any news on this?

**Comment by voelzmo** [comment]
Same observation was made in https://github.com/coredns/coredns/issues/6803
